### ISSUE TITLE
Can't get metafield value ID from metafield nor metafield values by metafield ID

### ISSUE THREAD

**Issue by kusmierz** [dreamcommerce/shop-appstore-lib#3]
I'm trying to fetch MetafieldValue (see [Metafield Values: get](https://developers.shoper.pl/developers/api/resources/metafield-values/get) reference) from given Metafield. But there is no MetafieldValue ID in response to do so (nor MetafieldValue's method to get MetafieldValue by Metafield ID).

In other words. I'm trying to list all metafields for given object type (`system` in this case):

```php
// ...
$resourceMetafield = new DreamCommerce\Resource\Metafield($client);
$resourceMetafieldValue = new DreamCommerce\Resource\MetafieldValue($client);

$object = 'system';
$metafields = $resourceMetafield->get($object);
```

Next I'm trying to iterate each and every metafield and get value for them:

```php
foreach ($metafields as $metafield) {
```

and here is the problem. `$metafield` is simple as this:

```php
array (
      'metafield_id' => '1175',
      'object' => 'SYSTEM',
      'key' => 'integration_version',
      'namespace' => 'DREAMCOMMERCE',
      'description' => 'asdfvgbfffffn',
      'type' => '3',
),
```

In other words, I can't call `$resourceMetafieldValue->get($metafieldValueId)`, as I don't have `$metafieldValueId`. I can't get [MetafieldValue](https://developers.shoper.pl/developers/api/resources/metafield-values/get) by MetafieldId either, 'cause there is no such as method.

It used to work, as long as `metafield_id` has the same value as `$metafieldValueId`, but it doesn't work any longer for us (1169 is our `$metafieldValueId` for this specific metafield with id 1175).

**Comment by Kyniu** [comment]
@er1z @jedi-pl could you look at this?

**Comment by er1z** [comment]
AFAIR, for this particular resource, GET method needs two arguments: context and then - ID. I'll check your code, stay tuned.

**Comment by er1z** [comment]
Due to the storing method of Metafields values, value has different value from Metafield.

This code should work for you:
```
    $resourceMetafield = new DreamCommerce\Resource\Metafield($client);
    $resourceMetafieldValue = new DreamCommerce\Resource\MetafieldValue($client);

    $key = 'metafield_key';
    $namespace = 'SYSTEM';
    $metafieldId = 2;

    $resource = new DreamCommerce\Resource\Metafield($client);
    $metafield = $resource->filters(['key'=>$key])->get($namespace, $metafieldId);

    if(!$metafield){
        die('no metafield with specified key');
    }

    $metafieldId = $metafield->metafield_id;

    $valueResource = new MetafieldValue($client);

    $value = $valueResource->filters([
        'metafield_id'=>$metafieldId
    ])->get();

    printf('metafield value: %s', $value[0]->value);
```

Feel free contacting me if you need further help.

**Comment by kusmierz** [comment]
I thought so... We copied your solution from [dreamcommerce/app-google-adwords-remarketing] (https://github.com/dreamcommerce/app-google-adwords-remarketing) and there you are using:

```php
$resultValue = $rMetafieldValue->get($resultMetafield->metafield_id);
```

So I think there is a bug, see: https://github.com/dreamcommerce/app-google-adwords-remarketing/blob/master/src/DreamCommerce/App/GoogleAdwordsRemarketing/Service/GoogleAdwords.php#L124

**Comment by deallas** [comment]
This is not a bug. An application Google Adwords Remarketing used library in version 0.1.10 (sorry for the omission file composer.lock) where function isCollection does not exist. Function GET with parameter for resource MetafieldValue should return ArrayObject.

**Comment by kusmierz** [comment]
I think would be easier to meet somewhere near to the Kraków Main Square for a beer or two ;)

What I was trying to say, in the code above, there is method get() on `DreamCommerce\Resource\MetafieldValue`, which, according to [the doc](https://developers.shoper.pl/developers/api/resources/metafield-values/get) has one param `id` (int), described as "metafield *_value_* identifier". BUT in this code, you passed a "metafield identifier", not "metafield value".

It will work as long as metafield has their value with the same id as its own. Now is it clear?

**Comment by deallas** [comment]
Oh sh... That's right !

**Comment by kusmierz** [comment]
no problem, you owe me a beer :)
